docs/migrate-to-nativecurlclient.md

```diff
@@ -0,0 +1,255 @@
+# Migrate from `Redmine\Client` to `Redmine\Client\NativeCurlClient`
+
+Since `php-redmine-api` v1.8.0 there is a new native cURL client
+`Redmine\Client\NativeCurlClient`. This guide will help you to migrate your code
+to this client.
+
+## 1. Use new client methods
+
+With the new interface `Redmine\Client\Client` there are now standarized methods
+for all clients. The new `Redmine\Client\NativeCurlClient` and the current `Redmine\Client` implementing this interface.
+
+### api() to getApi()
+
+Search in your code for the usage of `$client->api('issue')` and the magic
+getter like `$client->issue`. Then replace this calls with
+`$client->getApi('issue')`.
+
+```diff
+-$issue = $client->issue->show($issueId);
++$issue = $client->getApi('issue')->show($issueId);
+
+-$client->api('issue')->create($data);
++$client->getApi('issue')->create($data);
+```
+
+### getResponseCode() to getLastResponseStatusCode()
+
+Replace every call for `$client->getResponseCode()` with
+`$client->getLastResponseStatusCode()`.
+
+```diff
+-if ($client->getResponseCode() === 500)
++if ($client->getLastResponseStatusCode() === 500)
+{
+    throw new \Exception('Redmine call failed');
+}
+```
+
+### get() to requestGet()
+
+If you are using `$client->get()`, `$client->post()`, `$client->put()` or
+`$client->delete()` directly you will have to change your code. This methods
+parse a possible JSON or XML response but in future the parsing of the raw
+response body will be up to you.
```

Comment:
```suggestion
If you are using `$client->get()`, `$client->post()`, `$client->put()` or `$client->delete()` directly you will have to change your code. Currently, these methods parse the response (JSON or XML), but in future the parsing of the raw response body will be up to you.
```
(could we make all the paragraph a long line? Or is there a reason for having line breaks at the 80th chars?)

Reply:
I've noted that that shorter lines are much more useful in git diffs. But we could also got with long lines.